### PR TITLE
Adding network partitions for CLI

### PR DESCRIPTION
**Pull request by dinithis**
Adding network partitions for CLI

### CODE REVIEW

File: components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/commands/AddNetworkPartitionCommand.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    /**
+     * Information about the command
+     *
+     * @return The description of the command
+     */
+    @Override
+    public String getDescription() {
+        return "add network partition deployment";
```
Comment (imesh) [inline components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/commands/AddNetworkPartitionCommand.java]: May be we can change text "add" to "Add".

File: components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/RestCommandLineService.java [hunk 2]
```diff
@@ -1130,4 +1134,57 @@ private void printError(String message, Throwable e) {
         // Log error
         log.error(message, e);
     }
+
+    // This method helps to add network partition definitions
```
Comment (imesh) [inline components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/RestCommandLineService.java]: Shall we use Java method header format for this?
Thanks

**Comment by imesh** [comment]
Overall this pull request looks good, however I can see considerable amount of code formattings. It would be better to do code formatting in a different pull request as it is difficult to identify your changes. I have pointed out several points to fix. Please fix them and let us know.
Thanks

File: components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/commands/AddNetworkPartitionCommand.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+     * @param context The context assoicated with the Command Line Application
+     * @param args    The arguments for the command
+     * @return The status code
+     * @throws org.apache.stratos.cli.exception.CommandException if any errors occur when executing the command
+     */
+    @Override
+    public int execute(StratosCommandContext context, String[] args) throws CommandException {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Executing {} command...", getName());
+            logger.info("Get name" + getName());
```
Comment (swgkg) [inline components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/commands/AddNetworkPartitionCommand.java]: Info log available inside the isDebugEnabled() condition.Better to change that info log.

File: components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/RestCommandLineService.java [hunk 4]
```diff
@@ -1130,4 +1134,57 @@ private void printError(String message, Throwable e) {
...
+                }
+            };
+
+            NetworkPartitionBean[] partitions = new NetworkPartitionBean[networkPartitionsList.size()];
+            partitions = networkPartitionsList.toArray(partitions);
+
+            System.out.println("Network partitions found:");
+            CliUtils.printTable(partitions, networkPartitionMapper, "PartitionId", "Kubernetes Cluster Id", "Partitions");
+        } catch (Exception e) {
+            String message = "Error in listing network partitions" + e;
```
Comment (swgkg) [inline components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/RestCommandLineService.java]: We don't need to append exception to the message. We can have our own message.

Thanks

**Comment by dinithis** [comment]
Adding network partitions for CLI - changed suggested modifications

File: components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/RestCommandLineService.java [hunk 5]
```diff
@@ -293,44 +297,44 @@ public void listCartridges() throws CommandException {
...
+     */
+    public void listCartridgeGroups() throws CommandException {
+        try {
+            Type listType = new TypeToken<ArrayList<GroupBean>>() {
+            }.getType();
+            List<GroupBean> cartridgeGroupList = (List<GroupBean>) restClient.listEntity(ENDPOINT_LIST_CARTRIDGE_GROUPS,
+                    listType, "Cartridge Groups");
+
+            if ((cartridgeGroupList == null) || (cartridgeGroupList.size() == 0)) {
+                System.out.println("No cartridges found");
```
Comment (prabathabey) [inline components/org.apache.stratos.cli/src/main/java/org/apache/stratos/cli/RestCommandLineService.java]: Sys.outs need to be removed.

**Pull request closed without merging**